credoai/reporting/model_fairness.py

```diff
@@ -124,6 +123,110 @@ def plot_fairness(self):
                 self._plot_disaggregated_metrics(axes[1])
         return f
 
+    def _plot_fairness_metrics(self, ax):
+        # create df
+        df = self.module.get_fairness_results()
+        # add parity to names
+        df.index = [i+'_parity' if row['kind'] == 'parity' else i
+                    for i, row in df.iterrows()]
+        df = df['value']
+        df.index.name = 'Fairness Metric'
+        df.name = 'Value'
+        # plot
+        sns.barplot(data=df.reset_index(), 
+                    y='Fairness Metric', 
+                    x='Value',
+                    edgecolor='w',
+                    color = DEFAULT_COLOR,
+                    ax=ax)
+        self._style_barplot(ax)
+        
+    def _plot_disaggregated_metrics(self, ax):
+        # create df
+        sensitive_feature = self.module.sensitive_features.name
+        df =  self.module.get_disaggregated_performance(False) \
+                    .reset_index() \
+                    .melt(id_vars=sensitive_feature,
+                          var_name='Performance Metric',
+                          value_name='Value')
+        # plot
+        num_cats = len(df[sensitive_feature].unique())
+        palette = sns.color_palette('Purples', num_cats)
+        palette[-1] = [.4,.4,.4]
+        sns.barplot(data=df, 
+                    y='Performance Metric', 
+                    x='Value', 
+                    hue=sensitive_feature,
+                    palette=palette,
+                    edgecolor='w',
+                    ax=ax)
+        self._style_barplot(ax)
+        plt.legend(bbox_to_anchor=(1.01, 1.02))
+        
+    def _style_barplot(self, ax):
+        sns.despine()
+        ax.set_xlabel(ax.get_xlabel(), fontweight='bold')
+        ax.set_ylabel(ax.get_ylabel(), fontweight='bold')
+        # format metric labels
+        ax.set_yticklabels([format_label(label.get_text()) 
+                            for label in ax.get_yticklabels()])
+    
+
+class BinaryClassificationReporter(FairnessReporter):
```

Comment:
Made a child of FairnessReporter that separates the more specific BinaryClassification functionality from the general functionality